### PR TITLE
Add Stackoverflow Search Navigation

### LINKED ISSUE #429: [Feature Request] Search Navigator for StackOverflow
It would be nice to have keybindings to search through stackoverflow questions/ answers as well. Some ideas for keybinding:
j & k -> next / previous answer and replies
J & K -> next / previous answer
/ -> go to searchbar
c -> add comments
Any ideas for this? Thank you!

### PR DESCRIPTION
**Pull request by margunwa123**
Closes #429 

Checkpoints:
- [x] Stackoverflow seach result navigation
- [ ] Stackoverflow answers navigation

### CODE REVIEW

File: src/search_engines.js [hunk 1]
```diff
@@ -413,7 +413,7 @@ class GoogleSearch {
     observer.observe(container, {
       attributes: false,
       childList: true,
-      subtree: false,
+      subtree: true,
```
Comment (infokiller) [inline src/search_engines.js:416]: Is this change related to the PR?

File: src/options_page.js [hunk 2]
```diff
@@ -69,6 +69,7 @@ const OPTIONAL_PERMISSIONS_URLS = {
       '/*',
   ),
   'github': ['https://github.com/*'],
+  'stackoverflow': ['https://stackoverflow.com/*', 'https://www.stackoverflow.com/*'],
```
Comment (infokiller) [inline src/options_page.js:72]: Please change to `stack-overflow`, and for the JS variables below use `stackOverflow`

File: src/options_page.js [hunk 3]
```diff
@@ -141,6 +142,10 @@ class OptionsPageManager {
     github.addEventListener('change', () => {
       setSearchEnginePermission_(github);
     });
+    const stackoverflow = document.getElementById('stackoverflow');
```
Comment (infokiller) [inline src/options_page.js:145]: Please use stackOverflow

**Comment by infokiller** [comment]
Looks good overall, left minor style comments. Please also update the CHANGELOG and README.

**Review by infokiller — CHANGES_REQUESTED** [review]

**Comment by HerrAugust** [comment]
Hi everyone, I would be really interested in this feature as many others. Would it be possible to merge it? Or how can I help here?

Thanks!

**Comment by infokiller** [comment]
Hi @HerrAugust 
There are minor style changes needed (see my comments), and some testing would be good. I guess there may be some merge conflicts since master advanced since then

I may look into it later

**Pull request closed without merging**